src/js/components/Table/Paginated/PaginatedTable.tsx

```diff
@@ -102,20 +100,48 @@ const Filter = (
             : convertObjsToValueLabel(options as { [key: string]: string | React.ReactNode })
         }
       />
-    ) : (
+    );
+  };
+};
+
+const DateFilterType = (
+  fieldName: string,
+  label: string,
+  noDateLabel?: string,
+  yearPlaceholder?: string,
+  rangeMin?: string,
+  rangeMax?: string
+) => {
+  return ({ isLast }: { isLast: boolean }) => {
+    const ctx = React.useContext(FilterContext);
+
+    const onChange = (values) => {
+      if (values?.length > 0 || values?.kind) {
+        ctx.setActiveFilters({ ...ctx.activeFilters, [fieldName]: values });
+      } else {
+        delete ctx.activeFilters[fieldName];
+        ctx.setActiveFilters({ ...ctx.activeFilters });
+      }
+    };
+
+    return (
       <DateFilter
         mods={isLast ? '' : 'u-spaceRightSm'}
         onChange={onChange}
         name={fieldName}
         title={label}
-        noDateLabel={options as string}
+        noDateLabel={noDateLabel}
+        yearPlaceholder={yearPlaceholder}
+        rangeMin={rangeMin}
+        rangeMax={rangeMax}
       />
     );
   };
 };
 
 type PaginatedTableProps = React.FunctionComponent<PropTypes.InferProps<typeof propTypes>> & {
-  Filter: typeof Filter;
+  SelectFilter: typeof SelectFilterType;
```

Comment:
To really drive the type example home, *this* line shows us what the type is :) 
```
typeof SelectFilterType
```
the output of that expression is the actual type :)